sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala

```diff
@@ -123,6 +123,69 @@ case class ST_YMin(inputExpressions: Seq[Expression])
     copy(inputExpressions = newChildren)
   }
 }
+
+/**
+ * Test if returning Max X coordinate value.
+ *
+ * @param inputExpressions
+ */
+case class ST_XMax(inputExpressions: Seq[Expression])
+  extends UnaryGeometryExpression with CodegenFallback {
+  assert(inputExpressions.length == 1)
+
+
+  override protected def nullSafeEval(geometry: Geometry): Any = {
+    var coord:Array[Coordinate] = geometry.getCoordinates()
+    var maxval = Double.MinValue
+    for (point<-coord) {
+      if(point.getX()>maxval){
+        maxval = point.getX()
+      }
+    }
+    maxval
+
+  }
+
+  override def dataType: DataType = DoubleType
+
+  override def children: Seq[Expression] = inputExpressions
+
+  protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]) = {
+    copy(inputExpressions = newChildren)
+  }
+}
+
+/**
+ * Test if returning Min X coordinate value.
+ *
+ * @param inputExpressions
+ */
+case class ST_XMin(inputExpressions: Seq[Expression])
+  extends UnaryGeometryExpression with CodegenFallback {
+  assert(inputExpressions.length == 1)
+
+
+  override protected def nullSafeEval(geometry: Geometry): Any = {
+    var coord:Array[Coordinate] = geometry.getCoordinates()
+    var minval = Double.MaxValue
+    for (point<-coord) {
+      if(point.getX()<minval){
+        minval = point.getX()
+      }
+    }
+    minval
+
+  }
+
+  override def dataType: DataType = DoubleType
+
+  override def children: Seq[Expression] = inputExpressions
+
+  protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]) = {
+    copy(inputExpressions = newChildren)
+  }
+}
+
 case class ST_3DDistance(inputExpressions: Seq[Expression])
   extends BinaryGeometryExpression with CodegenFallback {
```

Comment:
Please carefully setup your IDE and carefully review your code submission. You are NOT supposed to change the format of the places that are NOT written by you.

Reply:
Am I supposed to make any changes in this file and commit again?

Reply:
Yes, use the original file and only add your code. Do NOT change the indents in other places. When you commit, you should review whether you have made changes to irrelevant places.

Reply:
I have copied the original Functions.scala file and added my functions at the end of the file.

Reply:
@kvpankaj1999 Like I said, you should review the change you made via Git. GitHub also provides such an interface. You still changed other places. This is mostly because your IDE has some auto formatter.

Reply:
I have removed my previous implementation of my functions from between and added them at the end of the file. When I compared the current file in master branch and my file in local there was no difference in the number of lines of code. Can you please let me know where exactly I have changed the file elsewhere.

Reply:
![image](https://user-images.githubusercontent.com/10948864/164951767-f7ad7dc3-f3ae-4dc0-8916-4c9f4253b011.png)

Please click "File changed" tab on this PR. You will see the spaces you added.

Reply:
I made the requested changes. Please let me know if there is still any issue.